modules/markup/renderer.go

```diff
@@ -44,18 +44,17 @@ type Header struct {
 
 // RenderContext represents a render context
 type RenderContext struct {
-	Ctx              context.Context
-	RelativePath     string // relative path from tree root of the branch
-	Type             string
-	IsWiki           bool
-	URLPrefix        string
-	Metas            map[string]string
-	DefaultLink      string
-	GitRepo          *git.Repository
-	ShaExistCache    map[string]bool
-	cancelFn         func()
-	TableOfContents  []Header
-	InStandalonePage bool // used by external render. the router "/org/repo/render/..." will output the rendered content in a standalone page
```

Comment:
`InStandalonePage` is for maintainable purpose, to avoid some one writes code `markup.Render(&markup.RenderContext{ })`  to render the external JS into current page, which would lead to XSS attack.

The caller must tell the render that "I am in a standalone page, I have set the security headers correctly, I want to render the external content".

Reply:
How to render standalone and embedded content differently, if differences are not limited to `iframe`'ing? PR #21098 uses `IsStandalonePage` to render raw PDF for `true` and an iframe with a PDF viewer HTML widget for `false`. After this PR is accepted, should #21098 reintroduce the property with a new meaning unrelated to security, literally is it a standalone page or not.